Fix @JSON Blade directive parsing with nested structures

elliota43 opened the pull request:
## Description

Fixes `@json` Blade directive incorrectly parsing complex expressions with nested structures.  Replaces `explode()` with tokenizer-based parser that respects nesting depth.

## Problem

The directive would split on commas inside nested arrays, closures, and method calls, causing parse errors.

**Example** (from original issue):

```php
@json([
    'items' => $helpers['benefit']['getAll']()->map(fn($item) => [
        'icon' => $item->icon,
        'title' => (string)$item->title,
        'description' => (string)$item->description
    ]),
    'translation' => '%:booking.benefits%'
])
```

would incorrectly split on the comma inside the closure.

## Solution

- Added `parseArguments()` method using PHP's tokenizer
- Tracks nesting depth (parentheses, brackets, braces)
- Only splits on commas at top level (`depth === 0`)
- Handles empty expressions (defaults to `null`)

Fully backwards compatible - defaults to same behavior as before.

Fixes #56331

elliota43 committed:
Fix @json Blade directive parsing with nested structures

Replaces `explode()` with tokenizer-based parser  that respects nested arrays, closures, and function calls.  Only splits on commas at the top level, preserving commas inside nested structure.

Example of previously broken case:
  @json([$items->map(fn($x) => ['a' => $x, 'b' => $x]), 'key' => 'value'])

This fix is backwards compatible - all existing @json usage continues
to work as before. The change only fixes previously broken cases with
complex nested structures.

Fixes #56331

elliota43 committed:
styleci/formatting

elliota43 committed:
style-ci

shaedrich commented on src/Illuminate/View/Compilers/Concerns/CompilesJson.php:
```diff
@@ -19,12 +19,90 @@ trait CompilesJson
+     * and other nested structures by using PHP's tokenizer.
+     *
+     * @param  string  $expression
+     * @return array
```
You can narrow this further down:

```suggestion
     * @return array{0?: string, 1?: string, 2?: string}
```

shaedrich replied:
you're welcome!

elliota43 committed:
refactoring `compileJson()`

- simplifying `compileJson()`
- adds more in depth docblock to `parseArguments()`'s return value

elliota43 committed:
typo in method name
